### PR TITLE
[IOTDB-6067] Pipe: Improve the stability of iotdb-thrift-connector-v2 during fault tolerance (avoid OOM)

### PR DESCRIPTION
**Pull request by DanielWang2035**
Currently, iotdb-thrift-connector-v2 will keep retrying when the connection to receiver is lost, and the failure won't be known by PipeAgent.
Fix:
- move retry threads into one thread pool
- if retry count exceeds limit, set the pipe status in current datanode to STOPPED.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/task/PipeTaskAgent.java [hunk 1]
```diff
@@ -315,6 +318,38 @@ private void dropAllPipeTasksWithoutLock() {
...
+                    .getRuntimeMeta()
+                    .getConsensusGroupId2TaskMetaMap()
+                    .values()
+                    .forEach(
+                        pipeTaskMeta -> {
+                          Iterator<PipeRuntimeException> exceptionIterator =
+                              pipeTaskMeta.getExceptionMessages().iterator();
+                          while (exceptionIterator.hasNext()) {
+                            if (exceptionIterator.next() instanceof PipeRuntimeCriticalException) {
+                              stopPipe(
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/task/PipeTaskAgent.java:345]: check before stop

Comment (DanielWang2035) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/task/PipeTaskAgent.java]: changed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java [hunk 2]
```diff
@@ -67,6 +69,11 @@ public PipeConnectorSubtask(
     super(taskID);
     this.inputPendingQueue = inputPendingQueue;
     this.outputPipeConnector = outputPipeConnector;
+    if (outputPipeConnector instanceof IoTDBThriftConnectorV2) {
+      retryPendingQueue = ((IoTDBThriftConnectorV2) outputPipeConnector).getRetryPendingQueue();
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java:73]: Try not to modify this class

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/v2/IoTDBThriftConnectorV2.java [hunk 3]
```diff
@@ -76,6 +80,13 @@ public class IoTDBThriftConnectorV2 implements PipeConnector {
 
   private static final CommonConfig COMMON_CONFIG = CommonDescriptor.getInstance().getConfig();
 
+  private final BoundedBlockingPendingQueue<Event> retryPendingQueue =
+      new BoundedBlockingPendingQueue<>(1024);
+
+  private static final ExecutorService transferRetryExecutor =
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/v2/IoTDBThriftConnectorV2.java:86]: double check init

**Review by SteveYurongSu — APPROVED** [review]

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**